docs/_ext/validating_code_block.py

```diff
@@ -22,28 +25,58 @@ class ValidatingCodeBlock(CodeBlock):
         'type-name': directives.unchanged,
     }
     option_spec.update(CodeBlock.option_spec)
-    skip_validation = (os.getenv('SPHINX_SKIP_CONFIG_VALIDATION') or 'false').lower() == 'true'
+
+    @cached_property
+    def configs(self) -> dict:
+        _configs = dict(
+            descriptor_path="",
+            skip_validation=False,
+            validator_path="bazel-bin/tools/config_validation/validate_fragment")
+        if os.environ.get("ENVOY_DOCS_BUILD_CONFIG"):
```

Comment:
Is writing to a file on access a common pattern for a `@[cached_]property`? Would it be possible to replace this with an explicit call?

Reply:
its not *writing* a file its *reading* it and caching the contents as a property of the class

there isnt so much a common pattern with `cached_property` as it has relatively recently been added to python (from django)

certainly you wouldnt want to do this with a not cached property, and probably not for something that actually persists

Reply:
> there isnt so much a common pattern with...

altho in so far as there is one - this is it - ie preventing repeated `io` calls for a class property

> probably you wouldnt want to do this with a not cached property,

in this case you would want generally want to be explicit - ie every time i call this, get me the current value

> and probably not for something that actually persists

i think this is more moot - eg get me a file path - but the first time you do - please populate it with `x` contents

Reply:
Ugh, indeed, I was thinking about https://github.com/envoyproxy/envoy/pull/15229/files#diff-b7bc9b4ee0614d05a36dbf7391fe7c20bc20f0f0a3dc67b86eb859139a333aa9R50 when writing the comment. Makes sense here, never mind me.